casbin/enforcer.cpp

```diff
@@ -162,8 +162,8 @@ bool Enforcer :: enforce(string matcher, Scope scope) {
 /**
  * Enforcer is the default constructor.
  */
-Enforcer* Enforcer :: NewEnforcer() {
-    Enforcer* e = new Enforcer;
+shared_ptr<Enforcer> Enforcer :: NewEnforcer() {
```

Comment:
How about also change `NewEnforcer` to C++ style constructor? The same question also found in `NewModel` and `NewEffector`

Reply:
Yeah, I agree we must switch to class constructors instead of methods for initialization.

Reply:
We can make this changes in any another PR, as it involves making lot of changes.